src/System.Private.CoreLib/shared/System/Threading/Mutex.Windows.cs

```diff
@@ -42,11 +34,11 @@ private void CreateMutexCore(bool initiallyOwned, string name, out bool createdN
             if (mutexHandle.IsInvalid)
             {
                 mutexHandle.SetHandleAsInvalid();
-#if PLATFORM_UNIX
+
                 if (errorCode == Interop.Errors.ERROR_FILENAME_EXCED_RANGE)
```

Comment:
I would expect that a name that is 10MB long is still going to fail with this error on Windows. Do we need similar special for the other synchronization primitives - to handle the case where the handle is really too long for the OS to handle? We should have a test for this. Also consider factoring out this error handling logic into a common method in WaitHandle.

Reply:
I tried the test with ```100k length``` . and it was not failing. should I increase the length or is there any other way to test it ?

Reply:
> Also consider factoring out this error handling logic into a common method in WaitHandle

Ignoring this because errors are handled differently in different functions. In functions like```CreateMutex``` we throw different exceptions for the different errors but for functions like ```OpenExistingWorker``` we return different worker status for different errors.